tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -317,15 +327,51 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn get_id_graph(who: &T::AccountId) -> Vec<(Identity, IdentityContext<T>)> {
+		fn insert_identity_with_limit(
+			owner: &T::AccountId,
+			identity: &Identity,
+			context: IdentityContext<T>,
+		) -> Result<(), DispatchError> {
+			IDGraphLens::<T>::try_mutate(owner, |len| {
+				let new_len = len.checked_add(1).ok_or(Error::<T>::IDGraphLenLimitReached)?;
+				if new_len > T::MaxIDGraphLength::get() {
```

Comment:
Why need to check again? I think previous line already checked.

Reply:
Not really. The previous line checks only for overflow, while the second checks against the defined limit.